[dtensor] PART 4: move remaining DTensor ops to core distributed

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88180
* #88179
* #88551
* #88178
* __->__ #88550
* #88177
* #88176
* #88549

This PR moves the view related DTensor ops to core distributed,
tests will be add in follow up PRs

part of https://github.com/pytorch/pytorch/issues/88838

wanchaol committed:
[dtensor] PART 4: move remaining DTensor ops to core distributed

This PR moves the view related DTensor ops to core distributed,
tests will be add in follow up PRs

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] PART 4: move remaining DTensor ops to core distributed"


This PR moves the view related DTensor ops to core distributed,
tests will be add in follow up PRs

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] PART 4: move remaining DTensor ops to core distributed"


This PR moves the view related DTensor ops to core distributed,
tests will be add in follow up PRs

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] PART 4: move remaining DTensor ops to core distributed"


This PR moves the view related DTensor ops to core distributed,
tests will be add in follow up PRs

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] PART 4: move remaining DTensor ops to core distributed"


This PR moves the view related DTensor ops to core distributed,
tests will be add in follow up PRs

[ghstack-poisoned]

fduwjj reviewed (approved):
The view op is indeed very complicated. Thanks! LGTM.

wanchaol committed:
Update on "[dtensor] PART 4: move remaining DTensor ops to core distributed"


This PR moves the view related DTensor ops to core distributed,
tests will be add in follow up PRs

part of https://github.com/pytorch/pytorch/issues/88838

[ghstack-poisoned]
